pkg/antctl/raw/supportbundle/command.go

```diff
@@ -523,8 +581,25 @@ func controllerRemoteRunE(cmd *cobra.Command, args []string) error {
 		defer f.Close()
 		io.Copy(f, reader)
 	}
-	if err := requestAll(agentClients, controllerClient, bar); err != nil {
-		return err
+	results := requestAll(agentClients, controllerClient, bar)
+	resultMap := downloadAll(agentClients, controllerClient, dir, bar, results)
+
+	resultStr := ""
+	var failedNodes []string
+	for k, v := range resultMap {
```

Comment:
I have a question for this part.
I am now invoking two separate commands in my scripts to collect antrea logs.
1. antctl supportbundle --controller-only -> If the command fails, will I still get an error after this change?
2. antctl supportbundle node_a node_b node_c -> If all the nodes fail instead of partial failures, will I get an error after this change?

Reply:
The current fix is not correct. There should be an error message to console in the end.  The two cases face the same problem, which is better to be shown to the user:
1. use a complex error message, include: is controller info failed? which nodes failed and why ?
2. in the `Failed Nodes` part of the output message, append error message to each line, eg:
```yaml
Failed Nodes:
- node-1: node is down
- node-2: ....
```
then , in the end. If the controller info is failed too, return the related error message. Or nil.

Reply:
For the output in the terminal, I don't have a strong preference and option 2 looks clear enough for me.

Actually, my question is more about the return value since I want to invoke these commands from another script.
It seems that it will output the error to terminal and return nil if there is nothing to be collected?
For these two commands, I prefer to return error if there is nothing that we can collect and return nil if we can collect partial bundle. What do you think?

Reply:
Sure, sounds pretty reasonable

Reply:
Updated. The latest output would look like:
<img width="1216" alt="image" src="https://user-images.githubusercontent.com/3939015/125742995-66a32c2c-9c36-473d-a092-7bf86846917c.png">

Reply:
Thanks for the change!